components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -784,44 +842,170 @@ private void addUserAttributesToCache(SessionDataCacheEntry sessionDataCacheEntr
+    }
+
+    private String analyzePromptParameter(OAuthMessage oAuthMessage, OAuth2Parameters params, String prompt) {
+
+        List promptsList = getSupportedPromts();
+        boolean contains_none = (OAuthConstants.Prompt.NONE).equals(prompt);
+
+        if (StringUtils.isNotBlank(prompt)) {
+            List requestedPrompts = getRequestedPromptList(prompt);
+            if (!CollectionUtils.containsAny(requestedPrompts, promptsList)) {
+                String message = "Invalid prompt variables passed with the authorization request";
+                return handleInvalidPromptValues(params, prompt, message);
+            }
+
+            if (requestedPrompts.size() > 1) {
+                if (requestedPrompts.contains(OAuthConstants.Prompt.NONE)) {
+
+                    String message = "Invalid prompt variable combination. The value 'none' cannot be used with others " +
+                            "prompts. Prompt: ";
+                    return handleInvalidPromptValues(params, prompt, message);
+
+                } else if (requestedPrompts.contains(OAuthConstants.Prompt.LOGIN) &&
+                        (requestedPrompts.contains(OAuthConstants.Prompt.CONSENT))) {
+                    oAuthMessage.setForceAuthenticate(true);
+                    oAuthMessage.setPassiveAuthentication(false);
+                }
+            } else {
+                if ((OAuthConstants.Prompt.LOGIN).equals(prompt)) { // prompt for authentication
+                    oAuthMessage.setForceAuthenticate(true);
+                    oAuthMessage.setPassiveAuthentication(false);
+                } else if (contains_none) {
+                    oAuthMessage.setForceAuthenticate(false);
+                    oAuthMessage.setPassiveAuthentication(true);
+                } else if ((OAuthConstants.Prompt.CONSENT).equals(prompt)) {
+                    oAuthMessage.setForceAuthenticate(false);
+                    oAuthMessage.setPassiveAuthentication(false);
+                }
+            }
+        }
+        return null;
+    }
+
+    private String handleInvalidPromptValues(OAuth2Parameters params, String prompt, String message) {
+        if (log.isDebugEnabled()) {
+            log.debug(message + " " + prompt);
+        }
+        OAuthProblemException ex = OAuthProblemException.error(OAuth2ErrorCodes.INVALID_REQUEST, message);
+        return EndpointUtil.getErrorRedirectURL(ex, params);
+    }
+
+    private List getRequestedPromptList(String prompt) {
+
+        String[] prompts = prompt.trim().split("\\s");
+        return Arrays.asList(prompts);
+    }
+
+    private List<String> getSupportedPromts() {
+        return Arrays.asList(OAuthConstants.Prompt.NONE, OAuthConstants.Prompt.LOGIN,
+                OAuthConstants.Prompt.CONSENT, OAuthConstants.Prompt.SELECT_ACCOUNT);
+    }
+
+    private String validatePKCEParameters(OAuth2ClientValidationResponseDTO validationResponse, String pkceChallengeCode, String pkceChallengeMethod) {
```

Comment:
Lets break this method to multiple lines since it exceeds 120 chars

Reply:
done